syncer/syncer.go

```diff
@@ -73,7 +77,18 @@ func (s *syncer) initializePeerMap() {
 	s.peerMap.PutPeers(peerStatuses)
 
 	for peerStatus := range s.syncPeerClient.GetPeerStatusUpdateCh() {
+		bestPeer := s.peerMap.BestPeer(nil)
+		peerBlockNum := peerStatus.Number
+
 		s.peerMap.Put(peerStatus)
+
+		// send a signal to watchSync if a new block arrives
+		if peerBlockNum > bestPeer.Number {
```

Comment:
I think bestPeer might be nil sometimes. (For instance, case of that node doesn't connect to any peer at first)

Reply:
This event is triggered when any peer sends a new event. If a peer has sent a new event that means that we have best peer inserted.
I added a guard just to be sure 380d48e091ae9f9daf8545f1e96847cf941e7937